DS/MainBinarySearch.java

```diff
@@ -0,0 +1,56 @@
+//optional program
+import java.util.InputMismatchException;
+import java.util.Scanner;
+class BinarySearch {
+	public int[] addElement(int[] list,int listSize,Scanner sc) {
+		for(int i=0;i<listSize;i++)
+			list[i]=sc.nextInt();
+		return list;
+	}
+	public int getItemLocation(int[] list,int item) {
+		int startingLocation=0,lastLocation=list.length-1,midLocation=(startingLocation+lastLocation)/2;
+		while(startingLocation<=lastLocation && list[midLocation]!=item) {
+			if(list[midLocation]<item) 
+				startingLocation=midLocation+1;
+			else
+				lastLocation=midLocation-1;
+			midLocation=(startingLocation+lastLocation)/2;
+		}
+		if(list[midLocation]==item)
+			return midLocation+1;
+		else
+			return 0;
+	}
+}
+public class MainBinarySearch {
+	public static void main(String args[]) {
+		BinarySearch binarySearchObject=new BinarySearch();
+		Scanner sc=new Scanner(System.in);
+		System.out.println("Enter size of list");
+		try {
+		int listSize=sc.nextInt();
+		int[] list=new int[listSize];int choise=-1;
+		System.out.println("Enter item");
+		list=binarySearchObject.addElement(list,listSize,sc);
+			while(true) {
+			System.out.println("Enter choise");
+			System.out.println("1:To Search an item\n2:Exit");
+				choise=sc.nextInt();
+			switch(choise) {
+				case 1:
+					System.out.println("Enter Item to be search");
+					int itemToBeSearch=sc.nextInt();
+					int getValue=binarySearchObject.getItemLocation(list,itemToBeSearch);
+					if(getValue==0)	
```

Comment:
What if I find a start element, Then too your method will return zero ( 0 ) . And here you are showing 'Item not found' in this case.
It should return '-1' if no element found. And check accordingly.